lightly/api/download.py

```diff
@@ -0,0 +1,192 @@
+import concurrent.futures
+import os
+import pathlib
+import shutil
+import threading
+import warnings
+from concurrent.futures import ThreadPoolExecutor
+from typing import List, Iterable, Union, Tuple, Dict
+
+import av
+import requests
+import tqdm
+from lightly.api import utils
+import PIL
+
+
+def download_image(url: str, session: requests.Session = None) -> PIL.Image.Image:
+    """Downloads an image from an url.
+
+    Args:
+        url: 
+            Url where image is stored.
+        session: 
+            Make request using the given session.
```

Comment:
This wording is a little ambiguous, as the session is not a make request. I would rather explain it e.g. as
`An existing session to perform the download. If it is not passed, a new one is created.`

Reply:
The same applies to other docstrings of sessions in this file.

Reply:
Or: "Session object to persist certain parameters across requests." (this is pretty much how the requests library describes a session).

Reply:
> If it is not passed, a new one is created.

This is not true